bazel/external/googleurl.patch

```diff
@@ -50,13 +50,74 @@ index 0cd36dc..8c4cbd4 100644
  #else
  #define STACK_UNINITIALIZED
  #endif
-
-# TODO(dio): Consider to remove the following patch when we have IDN-free optional build for URL
-# library from the upstream Chromium project. This is tracked in:
-# https://github.com/envoyproxy/envoy/issues/14743.
-
+@@ -365,8 +369,12 @@ inline constexpr bool AnalyzerAssumeTrue(bool arg) {
+ #endif  // defined(__clang_analyzer__)
+ 
+ // Use nomerge attribute to disable optimization of merging multiple same calls.
+-#if defined(__clang__) && __has_attribute(nomerge)
++#if defined(__clang__)
++#if defined(__has_attribute)
++#if __has_attribute(nomerge)
+ #define NOMERGE [[clang::nomerge]]
++#endif
++#endif
+ #else
+ #define NOMERGE
+ #endif
+@@ -392,8 +400,12 @@ inline constexpr bool AnalyzerAssumeTrue(bool arg) {
+ // See also:
+ //   https://clang.llvm.org/docs/AttributeReference.html#trivial-abi
+ //   https://libcxx.llvm.org/docs/DesignDocs/UniquePtrTrivialAbi.html
+-#if defined(__clang__) && __has_attribute(trivial_abi)
++#if defined(__clang__)
++#if defined(__has_attribute)
++#if __has_attribute(trivial_abi)
+ #define TRIVIAL_ABI [[clang::trivial_abi]]
++#endif
++#endif
+ #else
+ #define TRIVIAL_ABI
+ #endif
+@@ -401,8 +413,12 @@ inline constexpr bool AnalyzerAssumeTrue(bool arg) {
+ // Marks a member function as reinitializing a moved-from variable.
+ // See also
+ // https://clang.llvm.org/extra/clang-tidy/checks/bugprone-use-after-move.html#reinitialization
+-#if defined(__clang__) && __has_attribute(reinitializes)
++#if defined(__clang__)
++#if defined(__has_attribute)
++#if __has_attribute(reinitializes)
+ #define REINITIALIZES_AFTER_MOVE [[clang::reinitializes]]
++#endif
++#endif
+ #else
+ #define REINITIALIZES_AFTER_MOVE
+ #endif
+diff --git a/base/containers/checked_iterators.h b/base/containers/checked_iterators.h
+index b5fe925..553a772 100644
+--- a/base/containers/checked_iterators.h
++++ b/base/containers/checked_iterators.h
+@@ -217,7 +217,7 @@ using CheckedContiguousConstIterator = CheckedContiguousIterator<const T>;
+ 
+ }  // namespace base
+ 
+-#if defined(_LIBCPP_VERSION) && !defined(OS_NACL)
++#if defined(_LIBCPP_VERSION) && !defined(OS_NACL) && !defined(SKIP_CONTIGUOUS_ITERATOR_CPP17)
+ // Specialize both std::__is_cpp17_contiguous_iterator and std::pointer_traits
+ // for CCI in case we compile with libc++ outside of NaCl. The former is
+ // required to enable certain algorithm optimizations (e.g. std::copy can be a
+@@ -237,10 +237,6 @@ using CheckedContiguousConstIterator = CheckedContiguousIterator<const T>;
+ // [3] https://wg21.link/pointer.traits.optmem
+ namespace std {
+ 
+-template <typename T>
+-struct __is_cpp17_contiguous_iterator<::gurl_base::CheckedContiguousIterator<T>>
+-    : true_type {};
+-
```

Comment:
Deleted. Only difference between https://github.com/envoyproxy/envoy/pull/17958

Reply:
Any more info on this deletion? What impact does it have / are we tracking fixing this upstream somehow?

Reply:
It fixes it since the android error is trying to specialize something that doesn't exist. I'm surprise this builds at all though, maybe it's just a performance optimization?